pkg/registry/secret/etcd/etcd_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/rest/resttest"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/v1beta3"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools/etcdtest"
+)
+
+func newHelper(t *testing.T) (*tools.FakeEtcdClient, tools.EtcdHelper) {
+	fakeEtcdClient := tools.NewFakeEtcdClient(t)
+	fakeEtcdClient.TestIndex = true
+	helper := tools.NewEtcdHelper(fakeEtcdClient, v1beta3.Codec, etcdtest.PathPrefix())
+	return fakeEtcdClient, helper
+}
+
+func validNewSecret(name string) *api.Secret {
+	return &api.Secret{
+		ObjectMeta: api.ObjectMeta{
+			Name:      name,
+			Namespace: api.NamespaceDefault,
+		},
+		Data: map[string][]byte{
+			"test": []byte("data"),
+		},
+	}
+}
+
+func TestCreate(t *testing.T) {
+	fakeEtcdClient, helper := newHelper(t)
+	storage := NewStorage(helper)
+	test := resttest.New(t, storage, fakeEtcdClient.SetError)
+	secret := validNewSecret("foo")
+	secret.Name = ""
+	secret.GenerateName = "foo-"
+	test.TestCreate(
```

Comment:
> On Apr 28, 2015, at 12:32 PM, Paul Morie <notifications@github.com> wrote:
> 
> In pkg/registry/secret/etcd/etcd_test.go:
> 
> > +			Namespace: api.NamespaceDefault,
> > +		},
> > +		Data: map[string][]byte{
> > +			"test": []byte("data"),
> > +		},
> > +	}
> > +}
> > +
> > +func TestCreate(t *testing.T) {
> > +	fakeEtcdClient, helper := newHelper(t)
> > +	storage := NewStorage(helper)
> > +	test := resttest.New(t, storage, fakeEtcdClient.SetError)
> > +	secret := validNewSecret("foo")
> > +	secret.Name = ""
> > +	secret.GenerateName = "foo-"
> > +	test.TestCreate(
> Not your problem but I find the signature of TestCreate to be extremely unreadable
> 
Noted, people with time to come up with a better one are encouraged to provide pulls to improve it...
> —
> Reply to this email directly or view it on GitHub.
>